src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -229,46 +229,11 @@ public static void notifyChunksInRange(final World world, final boolean add, fin
         }
         centralChunk.markDirty();
         MineColonies.getNetwork().sendToAll(new UpdateChunkCapabilityMessage(centralChunk.getCapability(CLOSE_COLONY_CAP, null), centralChunk.x, centralChunk.z));
-
-        final int chunkX = centralChunk.x;
-        final int chunkZ = centralChunk.z;
-
-        final int range = Configurations.gameplay.workingRangeTownHallChunks;
-        final int buffer = Configurations.gameplay.townHallPaddingChunk;
-
-        final int maxRange = range * 2 + buffer;
-        @NotNull final File chunkDir = new File(DimensionManager.getWorld(0).getSaveHandler().getWorldDirectory(), CHUNK_INFO_PATH);
-        Utils.checkDirectory(chunkDir);
-
-        for(int i = chunkX - maxRange; i <= chunkX + maxRange; i++)
-        {
-            for (int j = chunkZ - maxRange; j <= chunkZ + maxRange; j++)
-            {
-                final boolean owning = i >= chunkX - range && j >= chunkZ - range && i <= chunkX + range && j <= chunkZ + range;
-                @NotNull final ChunkLoadStorage newStorage = new ChunkLoadStorage(id, ChunkPos.asLong(i, j), add, dimension, owning);
-                @NotNull final File file = new File(chunkDir, String.format(FILENAME_CHUNK, i, j, dimension));
-                if (file.exists())
-                {
-                    @Nullable final NBTTagCompound chunkData = loadNBTFromPath(file);
-                    final ChunkLoadStorage storage = new ChunkLoadStorage(chunkData);
-                    storage.merge(newStorage);
-                    if (storage.isEmpty())
-                    {
-                        file.delete();
-                    }
-                    else
-                    {
-                        saveNBTToPath(file, newStorage.toNBT());
-                        missingChunksToLoad++;
-                    }
-                }
-                else
-                {
-                    saveNBTToPath(file, newStorage.toNBT());
-                    missingChunksToLoad++;
-                }
-            }
-        }
+        final LoadColonyThread thread = new LoadColonyThread(centralChunk.x, centralChunk.z, id, dimension, add);
+        thread.start();
+        missingChunksToLoad = thread.getMissingChunksToLoad();
+        //todo store thread in ColonyManager and get value each time we need "missingChunksToLoad", if thread is null, then no need to get it
+        //todo on shut down do thread.join and then get missingChunksToLoad
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)